web3.js/src/publickey.ts

```diff
@@ -84,7 +92,18 @@ export class PublicKey extends Struct {
    * Return the base-58 representation of the public key
    */
   toBase58(): string {
```

Comment:
Let's not, and here's why!

String _literals_ in JavaScript are ‘interned.’ This means that these two strings:

```
const a = 'Hello';
const b = 'Hello';
```

…actually share the same address.

If it's worth space in their app (ie. because they rendered it) then this cache should add very little overhead while cutting down on CPU waste a lot.

Note: insert caveat here about how the cache grows without limits, the longer someone uses an app, even though the ‘screen’ that needed a given key might have long since disappeared.

Reply:
Yeah, I'm concerned about applications not having any control over the cache size. I think it'd be nice to have an opt out for use cases where long running applications handle millions of keys

Reply:
How about introducing a new memoized function for applications that are looking for a perf boost? Adding side effects to the most commonly used method might irk some developers. So we would have both `toBase58()` and `toBase58Memoized()`

Reply:
Aw… I can't do that to people. That would just stuff more things into `PublicKey` for folks to download, which they may or may not use but have to pay to download anyway.

I think we should either:

* be really opinionated, choose a tradeoff, and go with it, or
* do nothing and leave memoization up to the caller.

We could build an LRU cache of a fixed size, but oy…

Any idea of a public app that uses tons of keys that I could profile?